jenkins/library/src/main/groovy/redhat/openshift/addNode.groovy

```diff
@@ -0,0 +1,36 @@
+import hudson.model.AbstractBuild
+import hudson.model.Node
+import hudson.plugins.sshslaves.SSHLauncher
+import hudson.plugins.sshslaves.verifiers.NonVerifyingKeyVerificationStrategy
+import hudson.plugins.sshslaves.verifiers.SshHostKeyVerificationStrategy
+import hudson.slaves.ComputerLauncher
+import hudson.slaves.DumbSlave
+import hudson.util.VariableResolver
+import jenkins.model.Jenkins
+
+Jenkins jenkins = Jenkins.getInstance()
+
+Thread thread = Thread.currentThread()
+AbstractBuild build = thread.executable
+VariableResolver<String> resolver = build.getBuildVariableResolver()
+
+String nodeHostname = resolver.resolve("NODE_HOSTNAME")
+String nodeCredentialID = resolver.resolve("NODE_CREDENTIAL_ID")
+SshHostKeyVerificationStrategy strategy = new NonVerifyingKeyVerificationStrategy()
+ComputerLauncher nodeLauncher = new SSHLauncher(
```

Comment:
This is how you get sold on IntelliJ products....
![screenshot from 2017-03-30 10-38-08](https://cloud.githubusercontent.com/assets/7328370/24509684/01865acc-1535-11e7-95a2-b95d9517b020.png)